packages/ui-components/src/components/MetricCompareTable/MetricCompareTable.stories.tsx

```
@@ -16,8 +16,31 @@ const Template: ComponentStory<typeof MetricCompareTable> = (args) => (
   </Paper>
 );
 
+const RegionLinkComponent: RegionLinkComponentProp = ({
+  children,
+  region,
+}: {
+  children: React.ReactNode;
+  region: Region;
+}) => (
+  <a
+    href={`/states/${region.slug}`}
+    aria-label={region.shortName}
+    style={{ display: "block", textDecoration: "none", color: "initial" }}
+  >
+    {children}
+  </a>
+);
+
 export const Example = Template.bind({});
 Example.args = {
   regions: states.all,
   metrics: [MetricId.PI, MetricId.MOCK_CASES, MetricId.PASS_FAIL],
+  RegionLinkComponent,
```

Naively it seems like it would be simpler to just make this a `LinkComponent` (that takes an href, not a region) and have the compare table pass in the correct href and aria-label correctly based on the region itself.  Then to use the compare table you could hopefully just do:

```
<MetricCompareTable metrics={...} regions={...} LinkComponent={Link} />
```

or something like that.  And that pattern would work for any component, even if it's going to be linking to things besides regions.

Maybe I'm missing some of the rationale though?

---

That could work, but we will need to define the interface of the `LinkComponent` (in a similar way as the MUI Accordion component defines the API of acceptable `TransitionProps` https://mui.com/material-ui/transitions/#transitioncomponent-prop components) and we will need to pass the `RegionDB` instance (or at least the `RegionDB.getRegionUrl` method) to the `MetricCompareTable`.

The simplest interface for `LinkComponent` would be to receive `href`, `children` and maybe `aria-label`, but if later on we discover that we need additional properties (`onClick`, `className`, etc), we will need to update the interface and implementation of the compare table.

The benefit of defining the `LinkComponent` interface is that the implementation would be much simpler when implementing a client (in most cases we could just pass the next `Link` component) - for users using a non-compliant Link component (react-router) they could easily create a component that adapts the props.

Actually now that I'm writing this I think that passing the `RegionDB` instance and defining the `LinkComponent` interface is simpler and also more consistent (as we will need to pass the `RegionDB` to the map components anyways). 

I'm going to wait for more feedback, but I'm leaning towards the suggestion. Thanks @mikelehen !

---

Actually, does https://mui.com/material-ui/guides/routing/#global-theme-link solve this for us?  It looks like maybe we can configure the link component through the MUI theme?

If that doesn't work, I had one other related thought to share.  I keep being tempted to leverage ReactContext more given we are running into more "common things that components might need."  So we could consider wrapping everything up like:

```
<ActNowContext
  metricCatalog={...}
  regionDb={...}
  LinkComponent={...}>
  
   ... now you can use any of our components
   
</ActNowContext>
```

And FWIW- `LinkComponent` could default to a dummy version that uses `<a>` tags directly. 🤷